### PR TITLE
Multimodal add audio gen tools, eleven labs

### PR DESCRIPTION
**Pull request by anuragts**
## Description

- New eleven labs audio agent with cookbook example and playground agent

Please check the options that are relevant:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Model update
- [ ] Infrastructure change

## Checklist

- [x] My code follows Phidata's style guidelines and best practices
- [x] I have performed a self-review of my code
- [x] I have added docstrings and comments for complex logic
- [x] My changes generate no new warnings or errors
- [x] I have added cookbook examples for my new addition (if needed)
- [x] I have updated requirements.txt/pyproject.toml (if needed)
- [x] I have verified my changes in a clean environment

## Additional Notes

Include any deployment notes, performance implications, or other relevant information:

### CODE REVIEW

File: cookbook/agents/46_generate_audio.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+from phi.model.openai import OpenAIChat
+from phi.tools.eleven_labs_tools import ElevenLabsTools
+
+audio_agent = Agent(
+    model=OpenAIChat(id="gpt-4o"),
+    tools=[ElevenLabsTools()],
+    description="You are an AI agent that can generate audio using the ElevenLabs API.",
+    instructions=[
+        "When the user asks you to generate audio, use the `generate_audio` tool to generate the audio.",
+        "You'll pass the text to generate audio from to the tool.",
```
Comment (dirkbrnd) [inline cookbook/agents/46_generate_audio.py:11]:
```suggestion
        "You'll generate the appropriate prompt to send to the tool to generate audio.",
```

File: phi/tools/eleven_labs_tools.py [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+
+class ElevenLabsTools(Toolkit):
+    def __init__(
+        self,
+        api_key: Optional[str] = None,
+        save_path: str = "audio_generations",
+        voice_id: str = "21m00Tcm4TlvDq8ikWAM",
+        model_id: str = "eleven_multilingual_v2",
+        output_format: str = "mp3_44100_64",
```
Comment (dirkbrnd) [inline phi/tools/eleven_labs_tools.py:25]: Lets make this a string literal with all the possible options, so it is easier to pick

File: phi/tools/eleven_labs_tools.py [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+except ImportError:
+    raise ImportError("`elevenlabs` not installed. Please install using `pip install elevenlabs`")
+
+
+class ElevenLabsTools(Toolkit):
+    def __init__(
+        self,
+        api_key: Optional[str] = None,
+        save_path: str = "audio_generations",
+        voice_id: str = "21m00Tcm4TlvDq8ikWAM",
```
Comment (dirkbrnd) [inline phi/tools/eleven_labs_tools.py:23]: I assume this is an ElevenLabs thing?

File: phi/tools/eleven_labs_tools.py [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+        self.output_format = output_format
+
+        if not path.exists(self.save_path):
+            makedirs(self.save_path, exist_ok=True)
+
+        self.register(self.generate_audio)
+
+    def generate_audio(self, agent: Agent, prompt: str) -> str:
+        """
+        Use this function to generate audio from a text prompt.
```
Comment (dirkbrnd) [inline phi/tools/eleven_labs_tools.py:45]:
```suggestion
        Use this function to generate audio from a text prompt. The audio is stored to a file in the target_directory.
```

File: phi/tools/eleven_labs_tools.py [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+    from elevenlabs import ElevenLabs  # type: ignore
+except ImportError:
+    raise ImportError("`elevenlabs` not installed. Please install using `pip install elevenlabs`")
+
+
+class ElevenLabsTools(Toolkit):
+    def __init__(
+        self,
+        api_key: Optional[str] = None,
+        save_path: str = "audio_generations",
```
Comment (dirkbrnd) [inline phi/tools/eleven_labs_tools.py:22]:
```suggestion
        target_directory: str = "audio_generations",
```

File: phi/tools/eleven_labs_tools.py [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+    def generate_audio(self, agent: Agent, prompt: str) -> str:
+        """
+        Use this function to generate audio from a text prompt.
+
+        Args:
+            prompt (str): Text to generate audio from.
+        Returns:
+            str: Return the path to the generated audio file.
+        """
+        client = ElevenLabs(api_key=self.api_key)
```
Comment (dirkbrnd) [inline phi/tools/eleven_labs_tools.py:52]: Lets establish the client in the constructor?

File: phi/tools/eleven_labs_tools.py [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+
+        try:
+            audio_generator = client.text_to_speech.convert(
+                voice_id=self.voice_id,
+                model_id=self.model_id,
+                text=prompt,
+                output_format=self.output_format,
+            )
+
+            output_filename = f"{uuid4()}.mp3"
```
Comment (dirkbrnd) [inline phi/tools/eleven_labs_tools.py:62]: The suffix will depend on the output format right?

File: phi/tools/eleven_labs_tools.py (hunk 3 above)
Comment (anuragts) [reply phi/tools/eleven_labs_tools.py:23]: yes

File: phi/tools/eleven_labs_tools.py (hunk 7 above)
Comment (anuragts) [reply phi/tools/eleven_labs_tools.py:62]: Ahh, right

File: phi/tools/eleven_labs_tools.py [hunk 8]
```diff
@@ -0,0 +1,72 @@
+"""
+pip install elevenlabs
+"""
```
[the remark this answers is not in the document]
Comment (manthanguptaa) [reply phi/tools/eleven_labs_tools.py:3]: nitpick: remove this `pip install elevenlabs` from this file

**Review by manthanguptaa — APPROVED** [review]

**Pull request merged**